server/parsec/components/user.py

```diff
@@ -698,3 +709,20 @@ async def api_tos_accept(
                 return tos_cmds.latest.tos_accept.RepNoTos()
             case UserAcceptTosBadOutcome.TOS_MISMATCH:
                 return tos_cmds.latest.tos_accept.RepTosMismatch()
+
+    @api
+    async def api_list_frozen_users(
+        self,
+        client_ctx: AuthenticatedClientContext,
+        req: authenticated_cmds.latest.list_frozen_users.Req,
+    ) -> authenticated_cmds.latest.list_frozen_users.Rep:
+        outcome = await self.list_frozen_users(client_ctx.organization_id, client_ctx.user_id)
+        match outcome:
+            case UserListFrozenUsersBadOutcome.ORGANIZATION_NOT_FOUND:
+                client_ctx.organization_not_found_abort()
+            case UserListFrozenUsersBadOutcome.AUTHOR_NOT_ALLOWED:
+                return authenticated_cmds.latest.list_frozen_users.RepAuthorNotAllowed()
+            case UserListFrozenUsersBadOutcome.AUTHOR_NOT_FOUND:
+                return authenticated_cmds.latest.list_frozen_users.RepAuthorNotFound()
+            case frozen_users:
```

Comment:
Move the happy case first (this is to stay consistent with the rest of the codebase, and also given internally Python turns the match in a chain of `if .. elif .. elif ...`, so we avoid comparing to all the error cases)

On top of that, avoid using untyped variable the case: given they  accept any type it may introduce bug if we add new error types.

```suggestion
            case list() as frozen_users:
```